mlflow/tensorflow/_autolog.py

```diff
@@ -52,11 +57,110 @@ def on_epoch_end(self, epoch, logs=None):
         if epoch % self.log_every_n_steps == 0:
             self.metrics_logger.record_metrics(logs, epoch)
 
-    def on_train_end(self, logs=None):  # pylint: disable=unused-argument
-        if self.log_models:
-            registered_model_name = get_autologging_config(
-                mlflow.tensorflow.FLAVOR_NAME, "registered_model_name", None
-            )
-            mlflow.keras.log_model(
-                self.model, artifact_path="model", registered_model_name=registered_model_name
-            )
+
+def extract_sample_tensor_as_numpy_input_example_slice(
+    next_input: Union[tensorflow.Tensor, np.ndarray]
+) -> np.ndarray:
+    """
+    Extracts first `INPUT_EXAMPLE_SAMPLE_ROWS` from the next_input, which can either be of
+    numpy array or tensor type.
+
+    :param next_input: an input of type `np.ndarray` or `tensorflow.Tensor`
+    :return: a slice (of limit `INPUT_EXAMPLE_SAMPLE_ROWS`)  of the input of type `np.ndarray`
+
+    Examples
+    --------
+    when next_input is nd.array:
+    >>> input_data = np.array([1, 2, 3, 4, 5, 6, 7, 8])
+    >>> extract_sample_tensor_as_numpy_input_example_slice(input_data)
+    array([1, 2, 3, 4, 5])
+
+
+    when next_input is tensorflow.Tensor:
+    >>> input_data = tensorflow.convert_to_tensor([1, 2, 3, 4, 5, 6])
+    >>> extract_sample_tensor_as_numpy_input_example_slice(input_data)
+    array([1, 2, 3, 4, 5])
+    """
+
+    input_feature_slice = None
+    if isinstance(next_input, tensorflow.Tensor):
+        input_feature_slice = next_input.numpy()[0:INPUT_EXAMPLE_SAMPLE_ROWS]
+    elif isinstance(next_input, np.ndarray):
+        input_feature_slice = next_input[0:INPUT_EXAMPLE_SAMPLE_ROWS]
+    else:
+        warnings.warn(
+            f"Tensorflow estimator doesn't support '{type(next_input)}' type for features"
+        )
+    return input_feature_slice
+
+
+def extract_sample_dict_or_tensor(
+    next_input: Union[Dict[str, Union[tensorflow.Tensor, np.ndarray]], tensorflow.Tensor]
+) -> Union[Dict[str, np.ndarray], np.ndarray]:
```

Comment:
Why do we include `tensorflow.Tensor` as a valid input here if there's a separate function that covers this case? It looks like we delegate to that method in the implementation of this method anyway: https://github.com/mlflow/mlflow/pull/5510/files#diff-5cf5c34a96d4329cbe33c8f4e8ab769cbaa5d176042838a9385554f0f36ba03cR126. Can the calling code call `extract_sample_tensor_as_numpy_input_example_slice` instead?

```suggestion
def extract_input_example_from_tensor_dict(
    input_features: Dict[str, Union[tensorflow.Tensor, np.ndarray]]
) -> Dict[str, np.ndarray]:
```